### PR TITLE
[WIP] Add x509-ext crate

### PR DESCRIPTION
**Pull request by npmccallum**
Don't merge this yet. It is a work in progress.

### CODE REVIEW

File: x509-ext/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+extern crate alloc;
+
+pub mod pkix;
+
+/// A trait expressing the OID associated with an extension.
+pub trait Typed {
+    /// The OID which identifies an extension.
+    const OID: ObjectIdentifier;
+}
```
Comment (tarcieri) [inline x509-ext/src/lib.rs:22]: This might be interesting to add to `const-oid`. There are a few other places it could be used.

Comment (npmccallum) [reply x509-ext/src/lib.rs]: Done.

File: const-oid/src/lib.rs [hunk 2]
```diff
@@ -241,3 +241,9 @@ impl fmt::Display for ObjectIdentifier {
         Ok(())
     }
 }
+
+/// A trait expressing the association of a type with an OID.
+pub trait Typed {
```
Comment (tarcieri) [inline const-oid/src/lib.rs:246]: This could probably use a more descriptive name, but I'm having trouble thinking of a good one

Comment (tarcieri) [reply const-oid/src/lib.rs:246]: How about `OidValue`?

**Comment by npmccallum** [comment]
This has been (almost) superseded by #414. Closing.

**Pull request closed without merging**